### PR TITLE
Add special handling for cf2 file fields in attachments check filter 

### LINKED ISSUE #3116: Issue with attachments does not work
## Do You Need Immediate Help?
No

## Is This A Feature Request?
No

## What Version Of Caldera Forms, WordPress and PHP Are You Using?
Short Debug Information
WordPress Version: 5.1.1PHP Version: 7.0.33MySQL Version: 5.5.5Caldera Forms Version: 1.8.3WP_DEBUG: 1
Full Debug Information
WordPress Version: 5.1.1

PHP Version: 7.0.33

MySQL Version: 5.5.5

Server Software: Apache/2

Your User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.121 Safari/537.36

Session Save Path:

Session Save Path Exists: No

Session Save Path Writeable: No

Session Max Lifetime: 1440

Opcode Cache:

Apc: No
Memcached: No
Redis: No
Object Cache:

Apc: No
Apcu: No
Memcache: No
Memcached: No
Redis: No
WPDB Prefix: wp_

WP Multisite Mode: No

WP Memory Limit: 40M

Currently Active Theme: Twenty Nineteen: 1.3

Plugins:

Breadcrumb NavXT: 6.2.1 Active
Caldera Forms: 1.8.3 Active
Caldera Forms Anti Spam: 0.1.2 Active
CF Post Formats: 1.3.1 Active
Codestar Framework: 1.0.1.1 Active
Contact Form 7: 5.1.1 Active
Duplicate Post: 3.2.2 Active
Enhanced Text Widget: 1.4.6 Active
Envato Market: 2.0.1 Active
Max Mega Menu: 2.5.3.2 Active
Slider Revolution: 5.4.8.2 Active
ThemetechMount Brivona Demo Content Setup: 1.0 Active
ThemeTechMount Extras for Brivona Theme: 1.1 Active
WPBakery Page Builder: 5.6 Active
WPFront Scroll Top: 2.0.1 Not Active
Yoast SEO: 9.7 Active

## Does Your Issue Persist When You Disable All Other Plugins and Switch To The Default Theme?
YES

## What Is The Unexpected Behaviour?
Not all attachments are hitting server and email. 
I have tried with different configuration and the issue persist. 
Example: I am adding 4 attachments named 1.docx, 2.docx, 3.docx, 4.docx. I did it multiple times. Never all 4 files were uploaded.  One time files 1.docx and 3.docx were uploade, the other time 1.docx and 4.docx, other time 1,2 and 4. 

When trying to upload rar file - just ONE file. I am receiving empty email. 

Can you fix that ?

## What PHP Errors Have You Logged While Reproducing This Bug?
No errors

## What JavaScript Errors Have You Seen While Reproducing This Bug?
No errors

### PR DESCRIPTION
**Pull request by Shelob9**
Resolves #3116

I tested with a field that allows multiple files and has attatch to mailer checked.
Screen Shot 2019-03-27 at 11 51 27 AM

Before the fix, only one file was attached, after the fix multiple files are attached. If only one is provided, it attaches one. if I disable the multiple option, it works as expected.

### CODE REVIEW

File: classes/email/filters.php [hunk 1]
```diff
@@ -41,42 +41,55 @@ public static function mail_attachment_check( $mail, $data, $form){
...
-                } else {
-                    $file = str_replace( $dir[ 'baseurl' ], $dir[ 'basedir' ], $file );
-                    if ( is_string( $file ) ) {
-						$files = explode(',', $file);
-						foreach($files as $attachment){
-							$attachment = ltrim( $attachment, " ");
-							if( file_exists($attachment) ){
-								$mail[ 'attachments' ][] = $attachment;
+
+                if( Caldera_Forms_Field_Util::is_cf2_field_type(Caldera_Forms_Field_Util::get_type($field,$form))){
```
Comment (Shelob9) [inline classes/email/filters.php]: @New0 I did not want to mess with other file field types so I did all of this in an if/else, but I feel like the changes on L47 also should go on L67, and I don't like the copypasta. But it works.

**Comment by Shelob9** [comment]
This build 1.8.4-b.2 includes these fixes and the fixes for #3106 (error with mutli-lingual sites) and the Elementor/ Beaver Builder conflicts.

[1.8.4-b.2.zip](https://github.com/CalderaWP/Caldera-Forms/files/3014170/1.8.4-b.2.zip)

**Comment by Shelob9** [comment]
Here is a beta build with this fix - [1.8.5-b.1.zip](https://github.com/CalderaWP/Caldera-Forms/files/3098266/1.8.5-b.1.zip)

**Comment by New0** [comment]
@Shelob9 I tested adding a cf2 file field to the contact form sample.
Set the cf2 file field to allow multiple files and add attachments. 
I load Five files to the file field.
I submitted the form multiple times (reloading the page between each submission).

With the same five files I had one missing from time to time and not the same.

This is the behavior related in issue #3116

**Comment by Shelob9** [comment]
Related PR from 1.8.4 #3074 for issue #3046

**Comment by Shelob9** [comment]
>@Shelob9 It worked as expected using multiple light files, I'm not sure we want to include the npm update and version update in this PR though.
@new0 Valid concern, I need to be more careful to avoid doing that in the future, but I think we should merge this and release a beta.

**Pull request merged**